src/noLifecycleCallRule.ts

```diff
@@ -0,0 +1,47 @@
+import { IRuleMetadata, RuleFailure } from 'tslint';
+import { AbstractRule } from 'tslint/lib/rules';
+import { CallExpression, isPropertyAccessExpression, SourceFile, SyntaxKind } from 'typescript';
+import { NgWalker } from './angular/ngWalker';
+import { isLifecycleHookMethod } from './util/utils';
+
+export class Rule extends AbstractRule {
+  static readonly metadata: IRuleMetadata = {
+    description: 'Disallows explicit calls to lifecycle hooks.',
+    options: null,
+    optionsDescription: 'Not configurable.',
+    rationale: "Explicit calls to lifecycle hooks could be confusing. Invoke lifecycle hooks is an Angular's responsability.",
+    ruleName: 'no-lifecycle-call',
+    type: 'functionality',
+    typescriptOnly: true
+  };
+
+  static readonly FAILURE_STRING = 'Avoid explicit calls to lifecycle hooks';
+
+  apply(sourceFile: SourceFile): RuleFailure[] {
+    return this.applyWithWalker(new ExpressionCallMetadataWalker(sourceFile, this.getOptions()));
+  }
+}
+
+export class ExpressionCallMetadataWalker extends NgWalker {
```

Comment:
Do we need to export this?

Reply:
Not necessarily, however, for some reason, we export the walker's in almost all rules.

Reply:
👍